src/openrct2/actions/StaffHireNewAction.cpp

```diff
@@ -201,10 +201,25 @@ GameActions::Result StaffHireNewAction::QueryExecute(bool execute) const
         newPeep->TrousersColour = colour;
 
         // Staff energy determines their walking speed
-        newPeep->Energy = 0x60;
-        newPeep->EnergyTarget = 0x60;
-        newPeep->StaffMowingTimeout = 0;
+        switch (gCheatsSelectedStaffSpeed)
+        {
+            case 0:
+                newPeep->Energy = CHEATS_STAFF_FREEZE_SPEED;
+                newPeep->EnergyTarget = CHEATS_STAFF_FREEZE_SPEED;
+                break;
+
+            case 1:
+                newPeep->Energy = CHEATS_STAFF_NORMAL_SPEED;
+                newPeep->EnergyTarget = CHEATS_STAFF_NORMAL_SPEED;
+                break;
+
+            default:
+                newPeep->Energy = CHEATS_STAFF_FAST_SPEED;
+                newPeep->EnergyTarget = CHEATS_STAFF_FAST_SPEED;
+                break;
+        }
```

Comment:
This needs to be re-worked. When the cheat is in-active it should just have the normal speed and only then should it check what the cheat overrides.

Reply:
Not entirely sure if I understand this correctly.

if gCheatsSelectedStaffSpeed is set to NONE, I still need to change energy to CHEATS_STAFF_NORMAL_SPEED in order for the peeps to go back to normal.

Hopefully the current implementation at least makes it clear what is going on.

Reply:
The default speed should be ordinarily set to normal, cheats override the value but also works for me when its set to default with `None`